File: apps/menu-service/menu/src/main/java/com/example/menu/service/MenuService.java
package com.example.menu.service;

import com.example.menu.dto.MenuItemRequest;
import com.example.menu.dto.UpdateMenuItemRequest;
import com.example.menu.model.MenuItem;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface MenuService {

    MenuItem createMenuItem(MenuItemRequest request, MultipartFile image);

    List<MenuItem> getAllMenuItems();

    MenuItem getMenuItemById(String id);

    List<MenuItem> getMenuItemsByCategory(String category);

    List<MenuItem> getAvailableMenuItems();

    MenuItem updateMenuItem(String id, UpdateMenuItemRequest request, MultipartFile image);

    void deleteMenuItem(String id);
}

File: apps/restaurant-user-frontend/src/services/menuService.js
import menuApi from "../api/menuApi";

export const getAllMenuItems = async () => {
  const response = await menuApi.get("/menu");
  return response.data;
};

export const getMenuItemById = async (id) => {
  const response = await menuApi.get(`/menu/${id}`);
  return response.data;
};

export const getMenuItemsByCategory = async (category) => {
  const response = await menuApi.get(`/menu/category/${category}`);
  return response.data;
};

export const getAvailableMenuItems = async () => {
  const response = await menuApi.get("/menu/available");
  return response.data;
};

export const createMenuItem = async (formData) => {
  const response = await menuApi.post("/menu", formData);
  return response.data;
};

export const updateMenuItem = async (id, formData) => {
  const response = await menuApi.put(`/menu/${id}`, formData);
  return response.data;
};

export const deleteMenuItem = async (id) => {
  const response = await menuApi.delete(`/menu/${id}`);
  return response.data;
};